src/main/java/com/applicaster/jwplayerplugin/JWPlayerUtil.java

```diff
@@ -55,22 +55,35 @@ public static PlaylistItem getPlaylistItem(Playable playable, Map pluginConfigur
 
     private static List<Caption> getCaptions(Playable playable) {
         List<Caption> captionList = new ArrayList<>();
+        ArrayList tracks = null;
 
-        if(playable instanceof APAtomEntry.APAtomEntryPlayable) {
-            List<LinkedHashMap<String, String>> sideCarCaptions =
-                    ((APAtomEntry.APAtomEntryPlayable) playable).getEntry()
-                            .getExtension("sideCarCaptions", ArrayList.class);
-
-            if(sideCarCaptions != null) {
-                for (int i = 0; i < sideCarCaptions.size(); i++) {
-                    LinkedHashMap<String, String> sideCarCaption = sideCarCaptions.get(i);
-                    Caption caption = new Caption.Builder().file(sideCarCaption.get("src"))
-                            .label(sideCarCaption.get("label")).build();
-                    captionList.add(caption);
+        if (playable instanceof APAtomEntry.APAtomEntryPlayable) {
+
+            LinkedHashMap textTracks = (LinkedHashMap) ((APAtomEntry.APAtomEntryPlayable) playable).getEntry().getExtensions().get("text_tracks");
+
+            if (textTracks != null) {
+                tracks = (ArrayList) textTracks.get("tracks");
+            }
+
+            if (tracks != null) {
+                for (int i = 0; i < tracks.size(); i++) {
+                    LinkedHashMap<String, String> sideCarCaption = (LinkedHashMap<String, String>) tracks.get(i);
+
+                    String src = sideCarCaption.get("source");
+                    String label = sideCarCaption.get("label");
+
```

Comment:
Please also obtain the "kind" parameter , and pass it to the CaptionBuilder.

Reply:
The setKind method is receiving  CaptionType enum which has the following params:
CAPTIONS,
CHAPTERS,
THUMBNAILS;

Here is the caption Data:
```
tracks: [
{
label: "Românesc",
kind: "subtitles",
type: "vtt",
language: "ro",
source: "https://mediafiles.cms.protvplus.ro/files/2019/08/29/a30b6e79-5b82-4e21-a6c7-607a8aa672dc-vtt_subtitles.vtt"
}
]
```

What are the kinds the data will hold and which kind will fit the CaptionType enum parameters?